_posts/2021-11-18-keras-updates/keras-updates.Rmd

```diff
@@ -0,0 +1,159 @@
+
+It is just that quick growth that has created, on the R side, the need for extensive low-level refactoring and enhancements. (Of course, the user-facing new functionality itself also had to be implemented!)
+
+Before we get to the promised highlights, a word on how we think about Keras.
+
+## Have your cake and eat it, too: A philosophy of (R) Keras
+
+If you've used Keras in the past, you know what it's always been intended to be: a high-level library, making it easy (as far as such a thing *can* be easy) to train neural networks in R. Actually, it's not just about *easy*. Keras would like to enable users to write natural-feeling, idiomatic-looking code. This, to a high degree, is achieved by its allowing for object composition though the pipe operator; it is also a consequence of its abundant wrappers, convenience functions, and functional (stateless) semantics.[^3]
+
+[^3]: Provided stateless semantics is possible, in some context. This is anything but a matter of course, with an object-oriented language like Python in the background.
+
+However, due to the way TensorFlow and Keras have developed on the Python side -- referring to the big architectural and semantic changes between versions 1.x and 2.x, first comprehensively characterized on this blog [here](https://blogs.rstudio.com/ai/posts/2019-10-08-tf2-whatchanges/) -- it has become more challenging to provide, to the R user, all of the functionality available on the Python side. In addition, maintaining compatibility with several versions of Python TensorFlow -- something R Keras has always been doing -- by necessity gets more and more challenging, the more wrappers and convenience functions you add.
+
+So this is where we complement the above "make it R-like and natural, where possible" with "make it easy to port from Python, where necessary". With the new low-level functionality, you won't have to wait for R wrappers to make use of Python-defined objects. Instead, Python objects may be sub-classed directly from R; and any additional functionality you'd like to add to the subclass is defined in a Python-like syntax. What this means, concretely, is that translating Python code to R has become a lot easier. We'll catch a glimpse of this in the second of our three highlights.
+
+## New in Keras 2.6/7: Three highlights
+
+Among the many new capabilities added in Keras 2.6 and 2.7, we quickly introduce three of the very most important.
+
+-   *Pre-processing layers* significantly help to streamline the training workflow, integrating data manipulation and data augmentation.
+
+-   The ability to subclass Python objects (already alluded to several times) is the new low-level magic that, at the same time, is available to the `keras` user *and* is powering many user-facing enhancements underneath.
+
+-   Recurrent neural network (RNN) layers gain a new cell-level API.
```

Comment:
Is there a reason you italicize _Pre-processing layers_ in the first item, but not RNN layers here? Maybe each bullet could have something emphasized?